### PR TITLE
Converts BraveryPanel into redux component

### LINKED ISSUE #9454: Redux component - BraveryPanel
## Test plan

https://github.com/brave/browser-laptop/pull/9455#issue-235844791

----

**Describe the issue you encountered:**
Convert `BraveryPanel` into redux component.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #9454

Auditors: @bsclifton @bridiver

Test Plan:
- `npm run test -- --grep='Bravery Panel'`
- check if bravery panel is opened when you click on lion head
- check if you can toggle all checkboxes
- check if blocking numbers are correct

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/common/lib/braveryPanelUtil.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const getDisplayHost = (lastCommittedURL) => {
+  const parsedUrl = urlParse(lastCommittedURL)
+  if (parsedUrl.protocol === 'https:' || parsedUrl.protocol === 'http:') {
+    return parsedUrl.host
+  }
+
+  return lastCommittedURL
+}
+
+const getRedirectedResources = (redirectedResources) => {
```
Comment (bridiver) [inline app/common/lib/braveryPanelUtil.js]: I'm confused about the purpose of this

File: app/common/lib/braveryPanelUtil.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Immutable = require('immutable')
+
+// Utils
+const urlParse = require('../urlParse')
+
+const getDisplayHost = (lastCommittedURL) => {
```
Comment (bridiver) [inline app/common/lib/braveryPanelUtil.js]: this seems like something that should go in urlUtil

File: app/common/lib/braveryPanelUtil.js (hunk 1 above)
Comment (NejcZdovc) [inline app/common/lib/braveryPanelUtil.js]: You have `redirectedResources` in two level array and this function gets second level into one level.

Example:
``` 
{
  google: [
    'google.com',
    'google.si',
  ],
  fb: [
    'facebook.si'
  ]
}
```
into 
```
[
    'google.com',
    'google.si',
    'facebook.si'
]
```

File: app/common/lib/braveryPanelUtil.js (hunk 2 above)
Comment (NejcZdovc) [inline app/common/lib/braveryPanelUtil.js]: Moved

**Pull request merged**